ouroboros-consensus/src/unstable-consensus-testlib/Test/Util/Orphans/NoThunks.hs

```diff
@@ -32,11 +37,13 @@ instance NoThunks a => NoThunks (StrictMVar (IOSim s) a) where
       aMay <- unsafeSTToIO $ lazyToStrictST $ inspectMVar (Proxy :: Proxy (IOSim s)) (toLazyMVar mvar)
       noThunks ctxt aMay
 
-instance NoThunks a => NoThunks (StrictTVar (IOSim s) a) where
+instance NoThunks (StrictMVar (IOSim s) a) => NoThunks (Strict.StrictMVar (IOSim s) a) where
+  showTypeOf _ = "StrictMVar IOSim"
+  wNoThunks ctxt mvar = wNoThunks ctxt mvar
+
+instance NoThunks (StrictTVar (IOSim s) a) => NoThunks (Strict.StrictTVar (IOSim s) a) where
   showTypeOf _ = "StrictTVar IOSim"
-  wNoThunks ctxt tvar = do
-      a <- unsafeSTToIO $ lazyToStrictST $ inspectTVar (Proxy :: Proxy (IOSim s)) $ toLazyTVar tvar
-      noThunks ctxt a
+  wNoThunks ctxt tv = wNoThunks ctxt tv
```

Comment:
Aren't these definitions of `wNoThunks` self-recursive and will hence loop at runtime?

Reply:
Hmmm. Yeah they probably are. Tests pass fine though. It compiled on ghci so I went forward. Will look into it tomorrow (just need some intermediate cast functions)